File: DSA10005.py
from collections import defaultdict
def function():
    v, e = map(int, input().split())
    ls = list(map(int, input().split()))
    degp = defaultdict(int)
    degm = defaultdict(int)
    for i in range(0, 2*e, 2):
        degp[ls[i]] += 1
        degm[ls[i + 1]] += 1

    cnt = 0
    for i in range(1, v + 1):
        if degp[i] != degm[i]:
            cnt += 1
    print (1 if cnt == 0 else 0)



if __name__ == "__main__":
    for i in range(int(input())):
        function()

File: DSA10005.cpp
#include <iostream>
#include <map>
using namespace std;
int main () { 
    int t; cin >> t;
    while(t--) {
        int v, e; cin >> v >> e;
        map<int, int> degp;
        map<int, int> degm;

        for (int i = 0; i < e; i ++){
            int u, v; cin >> u >> v;
            degp[u] ++;
            degm[v] ++;
        }
        int cnt = 0;
        for (int i = 1; i < v; i++) {
            if (degp[i] != degm[i]) cnt ++;
        }
        cout << (cnt == 0? 1 : 0) << endl;
    }
}